File: web-service/src/main/java/Post/Graded.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Post;

import Notification.Notification;
import Registration.CourseOffering;
import core.Permission;

/**
 * used to record student grades for example an exam, quiz .. etc
 */
public class Graded extends AbstractPost {

	CourseOffering cource;
	int maxGrade, scaleFactor;

	@Override
	protected Notification generateNotification(String title, String message) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public Permission hasPermission() {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}

	@Override
	public Object[] search(String query) {
		throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
	}
}

File: web-service/src/models/post/Graded.ts
import { CourseOffering } from '../registration/course-offering';
import { AbstractPost } from './abstract-post';
import { Permission } from '../core/permission';

/**
 * used to record student grades for example an exam, quiz .. etc
 */
export class Graded extends AbstractPost {
	
	cource: CourseOffering;
	maxGrade: number;
	scaleFactor: number;

	generateNotification(title: string, message: string): Notification {
		return null;
	}

	public hasPermission(): Permission {
		return null;
	}

	public search(query: string): any[] {
		return null;
	}

	constructor() {
		super();
		this.maxGrade = 0;
		this.scaleFactor = 0;
	}
}